### PR TITLE
[MISC] Handle on start secret exception and remove stale test

### PR DESCRIPTION
**Pull request by dragomirp**
* Handle model error during _on_start hook
* Remove stale TLS test

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -1063,7 +1063,11 @@ def _on_start(self, event: StartEvent) -> None:
         if not self._can_start(event):
             return
 
-        postgres_password = self._get_password()
+        try:
+            postgres_password = self._get_password()
+        except ModelError:
+            logger.debug("_on_start: secrets not yet available")
+            postgres_password = None
```
Comment (dragomirp) [inline src/charm.py:1070]: Sometimes fails in CI. Eg: https://github.com/canonical/pgbouncer-operator/actions/runs/10086523896/job/27890280183#step:29:1933

File: tests/integration/test_tls.py [hunk 2]
```diff
@@ -188,50 +186,3 @@ async def test_tls_enabled(ops_test: OpsTest) -> None:
         for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
             assert await check_tls(ops_test, unit.name, enabled=False)
             assert await check_tls_patroni_api(ops_test, unit.name, enabled=False)
-
-
-@pytest.mark.group(1)
-@pytest.mark.skipif(
-    not os.environ.get("RESTART_MACHINE_TEST"),
-    reason="RESTART_MACHINE_TEST environment variable not set",
-)
```
Comment (dragomirp) [inline tests/integration/test_tls.py:197]: We are always skipping this test.

**Review by lucasgameiroborges — APPROVED** [review]

**Review by marceloneppel — APPROVED** [review]
LGTM! Thanks!

File: src/charm.py (hunk 1 above)
Comment (taurus-forever) [reply src/charm.py:1070]: @juditnovak please consider to report Juju if it is their issue OR fix our lib (no rush!).
In my mind, there should be no moment when secrets are not-yet available for the charm.
Correct me if I am wrong. Tnx!

**Review by taurus-forever — APPROVED** [review]

**Comment by juditnovak** [comment]
@taurus-forever 
I agree. 

Juju: 
 - I've opened a bug on this: https://bugs.launchpad.net/juju/+bug/2074369
    - Feel free to upvote!

Libs:
 - Safety measure should be introduced on our side: I opened issue https://github.com/canonical/data-platform-libs/issues/184

**Pull request merged**